src/main/java/org/ml_methods_group/algorithm/RMMR.java

```diff
@@ -0,0 +1,131 @@
+    private ExecutionContext context;
+
+    public RMMR() {
+        super(NAME, true);
+    }
+
+    @Override
+    protected List<Refactoring> calculateRefactorings(ExecutionContext context, boolean enableFieldRefactorings) throws Exception {
+        if (enableFieldRefactorings) {
+            // TODO: write to LOGGER or throw Exception? Change UI: disable field checkbox if only RMMR is chosen.
+            LOGGER.error("Field refactorings are not supported",
+                    new UnsupportedOperationException("Field refactorings are not supported"));
+        }
+        this.context = context;
+        init();
+        return runParallel(units, context, ArrayList::new, this::findRefactoring, AlgorithmsUtil::combineLists);
+    }
+
+    private void init() {
+        final EntitySearchResult entities = context.getEntities();
+        LOGGER.info("Init RMMR");
+        units.clear();
+        classEntities.clear();
+
+        classEntities.addAll(entities.getClasses());
+        units.addAll(entities.getMethods());
+        progressCount.set(0);
+
+        entities.getMethods().forEach(methodEntity -> {
+            List<ClassEntity> methodClass = entities.getClasses().stream()
+                    .filter(classEntity -> methodEntity.getClassName().equals(classEntity.getName()))
+                    .collect(Collectors.toList());
+            if (methodClass.size() != 1) {
+                LOGGER.error("Found more than 1 class that has this method");
+            }
+            methodsByClass.computeIfAbsent(methodClass.get(0), anyKey -> new HashSet<>()).add(methodEntity);
+        });
+    }
+
+    private List<Refactoring> findRefactoring(MethodEntity entity, List<Refactoring> accumulator) {
+        reportProgress((double) progressCount.incrementAndGet() / units.size(), context);
+        context.checkCanceled();
+        if (!entity.isMovable() || classEntities.size() < 2) {
+            return accumulator;
+        }
+        double minDistance = Double.POSITIVE_INFINITY;
+        double difference = Double.POSITIVE_INFINITY;
+        ClassEntity targetClass = null;
+        for (final ClassEntity classEntity : classEntities) {
+            final double distance = getDistance(entity, classEntity);
+            if (distance < minDistance) {
+                difference = minDistance - distance;
+                minDistance = distance;
+                targetClass = classEntity;
+            } else if (distance - minDistance < difference) {
+                difference = distance - minDistance;
+            }
+        }
+
+        if (targetClass == null) {
+            LOGGER.warn("targetClass is null for " + entity.getName());
+            return accumulator;
+        }
+        final String targetClassName = targetClass.getName();
+        double accuracy = (1 - minDistance) * difference;  // TODO: Maybe consider amount of entities?
+        if (!targetClassName.equals(entity.getClassName())) {
```

Comment:
No, it seems to me to be correct. We measure distance between CLASS and method, and distances between other classes and method. getDistance considers if it is calculating distance between CLASS and method so it will be counted as described in the article (without measuring method with itself which would be always 0 distance). And we suggest refactoring if there is less distance than with CLASS.